extension/intents/forms/formsContentScript.js

```diff
@@ -248,6 +244,51 @@ this.dictationContentScript = (function() {
     return `${startSlice}${spaceBefore}${snippet}${spaceAfter}${endSlice}`;
   }
 
+  function insertStringContentEditable(snippet, containerElement) {
+    if (window.getSelection) {
+        const sel = window.getSelection();
+        if (sel.getRangeAt && sel.rangeCount) {
+            let range = sel.getRangeAt(0);
+            // range.deleteContents(); TODO Maybe: overwrite selection
+
+            const modSnippet = prependAppendSpace(snippet, containerElement, range);
+
+            const snippetNode = document.createTextNode(modSnippet);
+            range.insertNode(snippetNode);
+
+            range = range.cloneRange();
+            range.setStartAfter(snippetNode);
+            range.collapse(true);
+            sel.removeAllRanges();
+            sel.addRange(range);
+        }
+    }
+  }
+
+  function prependAppendSpace(snippet, containerElement, range) {
+    let strAfter, strBefore;
+    const preRange = range.cloneRange();
+    const postRange = range.cloneRange();
+    preRange.collapse(true);
+    preRange.setStart(containerElement, 0);
+    if (preRange.toString().slice(-1) === "" || preRange.toString().slice(-1) === " ") {
```

Comment:
can you use `preRange.toString().endsWith(" ")` here?

Reply:
Done